### DISCUSSION TITLE
Customizable Import RAW data window.

### DISCUSSION THREAD

**Discussion by Coolguy19849** [audacity/audacity#4766] [Ideas & Feature Requests]
I think that for the next update to audacity, on the Import Raw Data window, there should be a way to specify custom encoding. Right now, there is only support for signed 8-bit, signed 16-bit, signed 24-bit, signed 32-bit, Unsigned 8-bit, etc. 
I would like to see support added in for 4-bit samples and any related sample types.

**Comment by rbdannenberg** [comment]
4 bit samples are often some kind of adaptive delta encoding. When Audacity was created there were a lot of formats and even raw audio files without headers where it was really useful to import raw data files, but this is only practical if the samples are not linear integers of a certain size. Otherwise, there are just too many cases to consider. One way to really specify a custom coding is open a binary file with Nyquist and convert the samples to floats.
